core/src/main/java/org/elasticsearch/cluster/routing/RoutingNodes.java

```diff
@@ -47,6 +40,12 @@
  */
 public class RoutingNodes implements Iterable<RoutingNode> {
 
+    private final static Random RANDOM;
+
+    static {
+        RANDOM = Randomness.create();
+    }
+
```

Comment:
When I left that last comment I didn't have you're other comments on my screen. I have no idea why though. Anyway, I like the `invoke(getRandom)` idea. Do you think we could do it if we see `RandomizedContext` on the classpath at all?